packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -16,6 +23,7 @@ import { VNodes } from './vnodes';
 import { checkVersionMismatch } from './check-version-mismatch';
 
 const signedTemplateMap: Map<LightningElementConstructor, Template> = new Map();
+const templateToApiVersionMap: Map<LightningElementConstructor, APIVersion> = new Map();
```

Comment:
It seems like we won't be able to trust type checking when it comes to things like `LightningElementConstructor`, since that interface may change over time. Is there a strategy to ensure type-checked compatibility across internal interfaces between `LOWEST_API_VERSION` and latest?

Reply:
I'm not following you. `LightningElementConstructor` is not likely to change anytime soon.

That said, we could also just set it to `any` or `function` or something, since I find I end up have to do an `as` cast anyway.

Reply:
It may be that any unconcern I had is unwarranted. As you point out, `LightningElementConstructor` is not likely to change. However, for posterity:

I was pointing out that, at line 26, the type of `Map<LightningElementConstructor, APIVersion>` is referenced. However, that's actually not a type we can trust, because the run-time semantics of that type will differ from the compile-time semantics.

Because the implementation of `LightningElementConstructor` can vary across versions of LWC, the `LightningElementConstructor` that is referenced here should not be thought of as an interface, but rather as a union of all the `LightningElementConstructor`s that might show up in the map with a corresponding API version. At runtime, that is. At compile time `LightningElementConstructor` will always refer to the interface that is implemented at the current commit. And type checking is done at run-time, leading to the disparity between run-time and compile-time semantics.

My question basically boiled down to: will there be a meta-`LightningElementConstructor` type that is unchanged across API versions, such that references to `LightningElementConstructor` in API-version-sensitive code will be provably correct? If we do not believe that `LightningElementConstructor` will be one of the things receiving breaking changes, then my question is probably moot.